test/index.test.js

```diff
@@ -1,94 +1,111 @@
-      connectionString
-    })
-    fastify.ready((err) => {
-      t.error(err)
-
-      t.ok(fastify.slonik)
-      t.ok(fastify.slonik.pool)
-      t.ok(fastify.slonik.connect)
-    })
-  })
-  test('fastify.sql', (t) => {
-    t.plan(2)
+test('Namespace should exist:', async t => {
+  const fastify = Fastify()
 
-    const fastify = Fastify()
-    t.teardown(() => fastify.close())
+  t.teardown(() => fastify.close())
 
-    fastify.register(fastifySlonik, {
-      connectionString
-    })
+  fastify.register(fastifySlonik, { connectionString })
+  await fastify.ready()
 
-    fastify.ready((err) => {
-      t.error(err)
+  t.test('fastify.slonik', async t => {
+    t.ok(fastify.slonik)
+    t.ok(fastify.slonik.pool)
+    t.ok(fastify.slonik.connect)
+    t.ok(fastify.slonik.query)
+    t.ok(fastify.slonik.transaction)
+    t.ok(fastify.slonik.exists)
+  })
 
-      t.ok(fastify.sql)
-    })
+  t.test('fastify.sql', async t => {
+    t.ok(fastify.sql)
   })
   t.end()
 })
 
-test('When fastify.slonik root namespace is used:', (t) => {
-  t.test('should be able to use make a query', t => {
-    t.plan(2)
+test('When fastify.slonik root namespace is used:', async t => {
+  const testName = 'foobar'
 
-    const fastify = Fastify()
-    t.teardown(() => fastify.close())
+  const fastify = Fastify()
 
-    fastify.register(fastifySlonik, {
-      connectionString
-    })
+  t.teardown(async () => {
+    const removeUser = fastify.sql`
+      DELETE FROM
+        users
+      WHERE
+        username=${testName};
+    `
+    await fastify.slonik.transaction(removeUser)
+    fastify.close()
+  })
+
+  fastify.register(fastifySlonik, { connectionString })
+  await fastify.ready()
 
-    fastify.ready(async (err) => {
-      t.error(err)
-      const queryString = fastify.sql`
-        SELECT 1 as one
-      `
+  t.test('should be able to make a query', async t => {
+    const queryString = fastify.sql`
+      SELECT 1 as one
+    `
+    const queryResult = await fastify.slonik.query(queryString)
+    const { rows: [{ one }] } = queryResult
+    t.same(one, 1)
+  })
 
-      const queryResult = await fastify.slonik.connect(connection => {
-        return connection.query(queryString)
-      })
+  t.test('should be able to make a transaction', async t => {
+    const queryString = fastify.sql`
+      INSERT INTO
+        users(username)
+      VALUES
+        (${testName})
+      RETURNING
+        *;
+    `
+    const queryResult = await fastify.slonik.transaction(queryString)
+    const { rows: [{ username }] } = queryResult
+    t.same(username, testName)
+  })
 
-      t.equal(queryResult.rows[0].one, 1)
-    })
+  t.test('should be able to make a exists query', async t => {
+    const queryString = fastify.sql`
+      SELECT
+        1
+      FROM
+        users
+      WHERE
+        username=${testName}
+    `
+    const queryResult = await fastify.slonik.exists(queryString)
+    t.ok(queryResult)
   })
   t.end()
```

Comment:
Could we remove `t.end` (here and elsewhere)? All of our tests are async, so `t.end` isn't necessary.